### PR TITLE
Versioned precompile script - drop 1.5 - support 1.6 and 1.7

### PR DESCRIPTION
**Pull request by t-bltg**
Fix https://github.com/JuliaPlots/Plots.jl/issues/3964.

I've tested bot generated precompilation scripts on ~~`1.5.4`,~~ `1.6.4` and `1.7.0`.

~~I tried to use `yml_path` as explained in https://timholy.github.io/SnoopCompile.jl/v1.5/bot/#Configure-the-bot-to-run-with-a-GitHub-Action-file-1, but it messes with the version in `SnoopCompile.yml` (`1` interpreted as `1.0` instead of `1.7`).~~

~~With `1.6` as LTS, should we drop `1.5` in `Project.toml` ?~~

### CODE REVIEW

**Comment by BeastyBlacksmith** [comment]
If we don't have a specific reason to drop 1.5 I wouldn't do so. I now set the tests to run on 1.5, 1.6, 1 and nightly and you may want to rebase this on master.
I'd just try if we can reuse the 1.6 precompile files on 1.5 otherwise we would have a reason.

**Comment by t-bltg** [comment]
> I'd just try if we can reuse the 1.6 precompile files on 1.5 otherwise we would have a reason.

Allright, I'll rework versions.

**Comment by t-bltg** [comment]
@BeastyBlacksmith, I'd prefer to use explicit minor versions instead of the majored `1` to avoid hitting https://github.com/JuliaPlots/Plots.jl/issues/3964 again.

As for reusing `1.6` precompilation files with `1.5` I'm afraid that won't be possible with the current configuration of `CompileBot`, we'd have to generate extrea precompilation files for `1.5`. Are you ok with this ?

File: src/precompile_includer.jl [hunk 1]
```diff
@@ -14,4 +14,16 @@ elseif !ismultios && !ismultiversion
         _precompile_()
     end
 else
+    @static if v"1.6.0-DEV" <= VERSION <= v"1.6.9"
```
Comment (BeastyBlacksmith) [inline src/precompile_includer.jl:17]: Don't you just have to change this line? Or will it be overwritten once the bot runs again?

Comment (t-bltg) [reply src/precompile_includer.jl:17]: Yes, it will be overwritten, this file is autogenerated.

File: deps/SnoopCompile/snoop_bot_config.jl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+using CompileBot
+
+botconfig = BotConfig(
+    "Plots",
+    version = [v"1.6"],
+    else_version = v"1.7",
```
Comment (BeastyBlacksmith) [inline deps/SnoopCompile/snoop_bot_config.jl:6]: I think I am in favor of putting `v"1.7"` also in `version`. IIUC then you might get a slower version of Plots on other versions, but can't get a broken version, or am I wrong?

Comment (t-bltg) [reply deps/SnoopCompile/snoop_bot_config.jl]: It looks like a safer approach.

**Comment by t-bltg** [comment]
I've replaced occurences of '1' in `.yml` files with lastest stable `1.7`.

I now need to regenerate `src/precompile_includer.jl`, and I think we'll be ok.

**Comment by t-bltg** [comment]
@BeastyBlacksmith, the failure on `1.5` is related to an old `GR_jll` artifact (missing `gr_polygonmesh3d` symbol in artifact `0.58.1`).

What do we do ?

**Comment by BeastyBlacksmith** [comment]
>  Is this a valid argument for dropping 1.5 ?

I'd say so

**Comment by t-bltg** [comment]
@BeastyBlacksmith, I know we have to spare github actions CI credits, but shouldn't `windows` and `macos` also be tested on `1.6` ? Testing only `ubuntu` for `nightly` is appropriate.

**Comment by BeastyBlacksmith** [comment]
Honestly I can't remember having seen a failure that is dependent on the OS __and__ the julia version.

**Comment by t-bltg** [comment]
> Honestly I can't remember having seen a failure that is dependent on the OS **and** the julia version.

Me neither, however, even if unlikely, it can still happen. Your call.

With `CI` passing, this PR is now ready.

**Review by BeastyBlacksmith — APPROVED** [review]
Let's see how it works out

**Comment by t-bltg** [comment]
Good, this looks consistent, and less likely to break `Plots` releases when `1.8.0` is out.

**Pull request merged**